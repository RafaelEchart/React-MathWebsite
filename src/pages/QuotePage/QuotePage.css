.authorClass {
    float: right;
}

.quoteClass {
    text-align: center;
}

.newQuoteIcon {
    float: right;
    font-size: 20px;
    margin-top: 10px;
    cursor: pointer;
}