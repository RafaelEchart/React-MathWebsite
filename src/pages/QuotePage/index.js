import React, { useState, useEffect } from 'react';
import './QuotePage.css';
import { FiRefreshCw } from 'react-icons/fi';

const QuotePage = () => {
  const [quote, setQuote] = useState(undefined);

  const getNewQuote = async () => {
    try {
      let newQuote = await fetch('https://random-math-quote-api.herokuapp.com/');
      newQuote = await newQuote.json();
      setQuote({
        quote: newQuote.quote,
        author: newQuote.author,
      });
    } catch (err) {
      setQuote({
        quote: 'There is no errors in math (This is an error message)',
        author: 'Rafael Echart',
      });
    }
  };

  useEffect(async () => {
    getNewQuote();
  }, []);

  return (
    <div className="homeContainer" id="custom-element">
      <div className="homeInnerContainer">
      <p>Math Quote..</p>
      {quote
      && <><h3 className="quoteClass">{quote.quote}</h3>
      <span className="authorClass">-{quote.author}</span><br/>
      <FiRefreshCw onClick={getNewQuote} className="newQuoteIcon"/></>}

      </div>
    </div>
  );
};
export default QuotePage;