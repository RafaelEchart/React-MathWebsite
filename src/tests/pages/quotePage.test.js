import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import QuotePage from '../../pages/QuotePage/index.js';

describe('Pages snapshots', () => {
  it('Should render QuotePage', async () => {
    const tree = renderer.create(<QuotePage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
