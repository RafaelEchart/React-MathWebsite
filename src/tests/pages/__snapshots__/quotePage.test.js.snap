// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages snapshots Should render QuotePage 1`] = `
<div
  className="homeContainer"
  id="custom-element"
>
  <div
    className="homeInnerContainer"
  >
    <p>
      Math Quote..
    </p>
  </div>
</div>
`;
